File: codigos/todo_list.py
# Lista de tarefas vazia: cada tarefa é um dict com id, título e status
tarefas = []

def mostrar_menu():
    print("\n=== Gerenciador de Tarefas ===")
    print("1. Criar tarefa")
    print("2. Listar tarefas")
    print("3. Concluir tarefa")
    print("4. Remover tarefa")
    print("5. Sair")

def listar_tarefas():
    if not tarefas:
        print("Nenhuma tarefa cadastrada.")
        return
    print("\nLista de tarefas:")
    for t in tarefas:
        status = "✓ Concluída" if t['concluida'] else "○ Pendente"
        print(f"ID: {t['id']} - {t['titulo']} ({status})")

def criar_tarefa():
    titulo = input("Digite o título da tarefa: ").strip()
    if titulo:
        nova_tarefa = {
            'id': len(tarefas) + 1,
            'titulo': titulo,
            'concluida': False
        }
        tarefas.append(nova_tarefa)
        print(f"Tarefa '{titulo}' criada com ID {nova_tarefa['id']}.")
    else:
        print("Título inválido.")

def concluir_tarefa():
    listar_tarefas()
    try:
        id_tarefa = int(input("Digite o ID da tarefa a concluir: "))
        for t in tarefas:
            if t['id'] == id_tarefa:
                t['concluida'] = True
                print(f"Tarefa ID {id_tarefa} marcada como concluída.")
                return
        print("ID inválido.")
    except ValueError:
        print("ID deve ser um número.")

def remover_tarefa():
    listar_tarefas()
    try:
        id_tarefa = int(input("Digite o ID da tarefa a remover: "))
        global tarefas
        tarefas = [t for t in tarefas if t['id'] != id_tarefa]
        print(f"Tarefa ID {id_tarefa} removida.")
    except ValueError:
        print("ID deve ser um número.")

# Loop principal
while True:
    mostrar_menu()
    opcao = input("Escolha uma opção: ").strip()
    
    if opcao == '1':
        criar_tarefa()
    elif opcao == '2':
        listar_tareas()
    elif opcao == '3':
        concluir_tarefa()
    elif opcao == '4':
        remover_tarefa()
    elif opcao == '5':
        print("Programa encerrado.")
        break
    else:
        print("Opção inválida. Tente novamente.")[web:13][web:15]

File: codigos/toDoList.js
// Lista de tarefas vazia: cada tarefa é um objeto com id, título e status
let tarefas = [];

function mostrar_menu() {
    console.log("\n=== Gerenciador de Tarefas ===");
    console.log("1. Criar tarefa");
    console.log("2. Listar tarefas");
    console.log("3. Concluir tarefa");
    console.log("4. Remover tarefa");
    console.log("5. Sair");
}

function listar_tarefas() {
    if (tarefas.length === 0) {
        console.log("Nenhuma tarefa cadastrada.");
        return;
    }
    console.log("\nLista de tarefas:");
    for (let t of tarefas) {
        let status = t.concluida ? "✓ Concluída" : "○ Pendente";
        console.log(`ID: ${t.id} - ${t.titulo} (${status})`);
    }
}

function criar_tarefa() {
    const readline = require('readline-sync');
    let titulo = prompt("Digite o título da tarefa: ") || "";
    titulo = titulo.trim();

    if (titulo) {
        let nova_tarefa = {
            id: tarefas.length + 1,
            titulo: titulo,
            concluida: false
        };
        tarefas.push(nova_tarefa);
        console.log(`Tarefa '${titulo}' criada com ID ${nova_tarefa.id}.`);
    } else {
        console.log("Título inválido.");
    }
}

function concluir_tarefa() {
    listar_tarefas();

    try {
        let id_tarefa = parseInt(prompt("Digite o ID da tarefa a concluir: ") || "0");
        for (let t of tarefas) {
            if (t.id === id_tarefa) {
                t.concluida = true;
                console.log(`Tarefa ID ${id_tarefa} marcada como concluída.`);
                return;
            }
        }
        console.log("ID inválido.");
    } catch (e) {
        console.log("ID deve ser um número.");
    }
}

function remover_tarefa() {
    listar_tarefas();

    try {
        let id_tarefa = parseInt(prompt("Digite o ID da tarefa a remover: ") || "0");
        tarefas = tarefas.filter(t => t.id !== id_tarefa);
        console.log(`Tarefa ID ${id_tarefa} removida.`);
    } catch (e) {
        console.log("ID deve ser um número.");
    }
}

// Loop principal
while (true) {
    mostrar_menu();
    let opcao = prompt("Escolha uma opção: ") || "";
    opcao = opcao.trim();

    if (opcao === '1') {
        criar_tarefa();
    } else if (opcao === '2') {
        listar_tarefas();
    } else if (opcao === '3') {
        concluir_tarefa();
    } else if (opcao === '4') {
        remover_tarefa();
    } else if (opcao === '5') {
        console.log("Programa encerrado.");
        break;
    } else {
        console.log("Opção inválida. Tente novamente.");
    }
}
